torch/fx/experimental/symbolic_shapes.py

```diff
@@ -4232,6 +4232,17 @@ def evaluate_guards_for_args(self, placeholders, args, *, ignore_static=True):
             return self.evaluate_guards_expression(code, args)
         return True
 
+    def drop_non_input_symint_guards(self, symints):
+        """
+        Removes guards that refer to non input symints
+        """
+        symints = {str(s) for s in symints}
```

Comment:
There's no reason to stringify these for the set? Symbols are hashable just fine? (Actually, is symints SymInt or sympy.Symbol?)

Reply:
`symints` are torch.SymInt; while, `self.guards[0].expr.free_symbols` are `sympy.core.symbol.Symbol`. I assume I need to convert them both to strings to compare them?

Reply:
Instead of what you're doing here, do `s.node.expr for s in symints if isinstance(s, torch.SymInt) and isinstance(s.node.expr, sympy.Symbol)`. Drop the SymInt test if you can guarantee these are SymInts. I can't actually tell from context.